pipeline_dp/aggregate_params.py

```diff
@@ -53,6 +53,12 @@ class NormKind(Enum):
     L2 = "l2"
 
 
+class PartitionSelectionStrategy(Enum):
```

Comment:
Idle thought: I wonder if having a PUBLIC_PARTITIONS option for PartitionSelectionStrategy would help with code clarity. We have many checks (`if public_partitions is not None`) for public_partitions that could be replaced with the enum. But that might require some refactoring. WDYT?

Reply:
it's an option, but it would require users to set this parameter directly:
  - to set `PUBLIC_PARTITIONS` when public partitions are used
  - to set some other method otherwise
  
 while it's more preferable that the users change this parameter only if needed (it's definitely a complicated parameter and it's hard for most users to understand educated decision)

Reply:
The users would not necessarily have to set it themselves though, we could set it for them once if public_partitions are supplied.